docs/architecture/deep-dives/passkeys/credentials.md

```diff
@@ -0,0 +1,104 @@
+- `privateKey` - Private key material, created when generating the public/private key pair.
+
+### Notable optional fields
+
+- `userHandle` - Binary data sequence that uniquely identifies the user account. This data is opaque
+  for anyone other than the RP. It is used to associate a credential with a specific user account.
+- `userName` - A human-readable name for the user account. This can be used to display the user
+  account in the authenticator's user interface (if supported).
+
+## Storage Modality
+
+### Client-Side Credentials
+
+The credential is stored in persistent storage embedded in the authenticator, client or client
+device. This is rapidly becoming a common storage mode, as it is the only way to support synced
+credentials across multiple devices. Client-side storage is required.
+
+Hardware security keys also support this storage mode, but they often have a limited capacity (e.g.
+the YubiKey 5 can hold up to 25 client-side credentials).
+
+### Server-Side Credentials
+
+The credential is encrypted by the authenticator and stored in a database managed by the RP. This is
+the traditional storage mode and is often used in 2FA flows.
+
+This enables the authenticator to have unlimited storage capacity for credentials, since the data is
+stored by the RP instead of by the authenticator. However, this means that a credential stored in
+this way must be retrieved from the RP before the authenticator can use it.
+
+## Discoverability
+
+### Non-Discoverable Credentials
+
+This is a credential whose `credentialId` must be provided in `allowCredentials` when calling
+`navigator.credentials.get` because it is not client-side discoverable. This is always the case when
+the credential is stored server-side, because the RP must first identify the user in order to
+discover the `credentialId` to supply in the `navigator.credentials.get` call.
+
+### Client-Side Discoverable Credentials
+
+:::info
+
+These credentials have been previously known as resident credentials or resident keys. Due to the
+phrases `ResidentKey` and `residentKey` being widely used in both the WebAuthn API and also in the
+Authenticator Model (e.g. in dictionary member names, algorithm variable names, and operation
+parameters) the usage of resident within their names has not been changed for backwards
+compatibility purposes.
+
+:::
+
+This credential is usable in authentication operations where the RP does not provide any
+`credentialIds`, meaning that the RP does not need to identify the user first.
+
+As a result, an authenticator that can handle discoverable credentials can create assertion
+signatures using just an `rpId`. This enables single-click authentication flows where the returned
+assertion data contains everything the RP needs to identify and authenticate the user.
+
+This implies that the source of the credential must be stored either in the authenticator or the
+client platform (see [Client Side Credentials](#client-side-credentials)).
```

Comment:
Would it be helpful to diagram a matrix of storage modality / discoverability for easy reference?  For me at least, it was very helpful to understand the "why" of discoverability when I understood the relationship with how the credentials are stored.